### PR TITLE
Toponaming move makEFace as makeElementFace and dependencies

### PR DESCRIPTION
**Pull request by bgbsww**

### CODE REVIEW

File: src/Mod/Part/App/FaceMaker.cpp [hunk 1]
```diff
@@ -172,12 +264,12 @@ TYPESYSTEM_SOURCE(Part::FaceMakerSimple, Part::FaceMakerPublic)
...
 std::string Part::FaceMakerSimple::getUserFriendlyName() const
 {
-    return {QT_TRANSLATE_NOOP("Part_FaceMaker","Simple")};
+    return std::string(QT_TRANSLATE_NOOP("Part_FaceMaker","Simple"));
 }
 
 std::string Part::FaceMakerSimple::getBriefExplanation() const
 {
-    return {QT_TRANSLATE_NOOP("Part_FaceMaker","Makes separate plane face from every wire independently. No support for holes; wires can be on different planes.")};
+    return std::string(QT_TRANSLATE_NOOP("Part_FaceMaker","Makes separate plane face from every wire independently. No support for holes; wires can be on different planes."));
```
Comment (chennes) [inline src/Mod/Part/App/FaceMaker.cpp:272]: @kaktusus can you look for this string in the translations? This looks suspect to me, I don't really expect a NOOP in this context. (@bgbsww nothing for you to worry about, we'll adjust in a later PR if this line is wrong).

File: src/Mod/Part/App/FaceMaker.h [hunk 2]
```diff
@@ -77,9 +86,9 @@ class PartExport FaceMaker: public BRepBuilderAPI_MakeShape, public Base::BaseCl
     virtual const TopoDS_Face& Face();
 
 #if OCC_VERSION_HEX >= 0x070600
-    void Build(const Message_ProgressRange& theRange = Message_ProgressRange()) override;
+    virtual void Build(const Message_ProgressRange& theRange = Message_ProgressRange());
```
Comment (chennes) [inline src/Mod/Part/App/FaceMaker.h:89]: Why the change from `override` to `virtual`? Isn't this overriding a base class method?

File: src/Mod/Part/App/TopoShape.h [hunk 3]
```diff
@@ -629,6 +655,58 @@ class PartExport TopoShape: public Data::ComplexGeoData
      */
     TopoShape &makeElementCompound(const std::vector<TopoShape> &shapes, const char *op=nullptr, bool force=true);
 
+    TopoShape &makeElementFace(const std::vector<TopoShape> &shapes,
+                        const char *op = nullptr,
+                        const char *maker = nullptr,
+                        const gp_Pln *pln = nullptr);
```
Comment (chennes) [inline src/Mod/Part/App/TopoShape.h:661]: Needs formatting cleanup -- does your IDE allow selective application of clang-format? I don't want to do the whole file, just all the new code.

File: src/Mod/Part/App/TopoShape.h [hunk 4]
```diff
@@ -629,6 +655,58 @@ class PartExport TopoShape: public Data::ComplexGeoData
...
+
+    /// Filling style when making a BSpline face
+    enum FillingStyle {
+        /// The style with the flattest patches
+        FillingStyle_Strech,
+        /// A rounded style of patch with less depth than those of Curved
+        FillingStyle_Coons,
+        /// The style with the most rounded patches
+        FillingStyle_Curved,
+    };
```
Comment (chennes) [inline src/Mod/Part/App/TopoShape.h:707]: Make this an `enum class` and drop the "FillingStyle_" prefixes.

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 5]
```diff
@@ -540,4 +548,250 @@ TopoShape::makeElementCompound(const std::vector<TopoShape>& shapes, const char*
...
+    else {
+        shapes.push_back(shape);
+    }
+    return makeElementFace(shapes, op, maker, pln);
+}
+
+TopoShape& TopoShape::makeElementFace(const std::vector<TopoShape>& shapes,
+                                      const char* op,
+                                      const char* maker,
+                                      const gp_Pln* pln)
```
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:573]: I hate this abbreviation, let's rename to "plane".

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 6]
```diff
@@ -540,4 +548,250 @@ TopoShape::makeElementCompound(const std::vector<TopoShape>& shapes, const char*
...
+        maker = "Part::FaceMakerBullseye";
+    }
+    std::unique_ptr<FaceMaker> mkFace = FaceMaker::ConstructFromType(maker);
+    mkFace->MyHasher = Hasher;
+    mkFace->MyOp = op;
+    if (pln) {
+        mkFace->setPlane(*pln);
+    }
+
+    for (auto& s : shapes) {
```
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:585]: I've been renaming all the single-char variables (I'm surprised the linter from our CI isn't complaining, actually... the one in my IDE yells about all of them).

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 7]
```diff
@@ -540,4 +548,250 @@ TopoShape::makeElementCompound(const std::vector<TopoShape>& shapes, const char*
...
+/**
+ *  Encode and set an element name in the elementMap.  If a hasher is defined, apply it to the name.
+ *
+ * @param element   The element name(type) that provides 1 one character suffix to the name IF <conditions>.
+ * @param names     The subnames to build the name from.  If empty, return the TopoShape MappedName.
+ * @param marker    The elementMap name or suffix to start the name with.  If null, use the
+ *                  elementMapPrefix.
+ * @param op        The op text passed to the element name encoder along with the TopoShape Tag
+ * @param _sids     If defined, records the sub ids processed.
+ *
```
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:637]:
```suggestion
 *
```

File: tests/src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 8]
```diff
@@ -161,4 +168,272 @@ TEST_F(TopoShapeExpansionTest, makeElementCompoundTwoCubes)
...
+    // Act
+    EXPECT_EQ(topoShape.getShape().Orientation(), TopAbs_FORWARD);
+    Part::TopoShape wire =
+        topoShape.splitWires(&inner, Part::TopoShape::SplitWireReorient::ReorientReversed);
+    // Assert
+    EXPECT_EQ(inner.size(), 1);
+    EXPECT_FLOAT_EQ(PartTestHelpers::getLength(wire.getShape()), 2 + 2 + 3 + 3);
+    EXPECT_FLOAT_EQ(PartTestHelpers::getLength(inner.front().getShape()), M_PI * R * 2);
+    EXPECT_EQ(wire.getShape().Orientation(), TopAbs_REVERSED);
+    for (Part::TopoShape ts : inner) {
```
Comment (chennes) [inline tests/src/Mod/Part/App/TopoShapeExpansion.cpp:429]:
```suggestion
    for (Part::TopoShape &ts : inner) {
```
??

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 6 above)
Comment (bgbsww) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:585]: <chuckle>  I remember swapping floppys to let the second stage of the compiler run, so the ancient unix/c shortnames don't always trigger me.  We'll get 'em fixed.

**Comment by kaktusus** [comment]
Since we are acting in a common cause, you know I'll be happy to help you.
:wink:

**Comment by kaktusus** [comment]
`Makes separate plane face from every wire independently. No support for holes; wires can be on different planes.` - This string is in crowdin
https://crowdin.com/editor/freecad/all/en-pl?view=comfortable&filter=basic&value=0#q=Makes%20separate%20plane%20face%20from%20every%20wire%20independently.%20No%20support%20for%20holes;%20wires%20can%20be%20on%20different%20planes.

it is a very old string
https://crowdin.com/project/freecad/activity-stream?lang=38&translation_id=6500350

:smile:

**Comment by chennes** [comment]
@kaktusus Do you know if it is appearing correctly in the interface?

**Comment by kaktusus** [comment]
This is already a more difficult task,
I will try to find this string in the GUI.
But it won't be as fast as it was a moment ago :wink: 

In the code, I see FaceMaker.
What tool calls this function?
